in_toto/hash_test.go

```diff
@@ -0,0 +1,33 @@
+package in_toto
+
+import (
+	"reflect"
+	"testing"
+)
+
+/*
+This function tests if all the hashing functions listed
+in the hash_func list inside hash.go file
+*/
+
+func TestHashFunctions(t *testing.T) {
+
+	hash_func := createList()
+	mapper := createMap()
+	expected := map[string]interface{}{
+		"sha256": "de31a0fb6adeb79b6017f39244ac52a4aea74548015c37d98f00a9e0e0914565",
+		"sha512": "9ebe8a6e2b814321588c920de4de007ad911edc7a6ca5e826fa5d0a99a217b79e13ecebe11bade0f6c3775f7023ca86cd1c58745ff98d1c8c1bb3fbc9c02608e",
+		"md5":    "b000762f953ddd0c704aaf0114d10556",
+	}
+
+	testMap := make(map[string]interface{})
+	for _, element := range hash_func {
+
+		result := mapper[element].Compute([]uint8("Hashing this string to test"))
+		testMap[element] = result
+	}
+
+	if !reflect.DeepEqual(testMap, expected) {
```

Comment:
Do you need deep equal here?

Reply:
Made all the requested changed.

Regarding deep equal, I think we will need it.

According to the description of on stack overflow

"it first checks if both maps are nil, then it checks if they have the same length before finally checking to see if they have the same set of (key, value) pairs."

I found similar description on the go source code 
https://golang.org/src/reflect/deepequal.go?s=2432:2441#L98

Please let me know if you still think deep equal is not apt. I can then write a function to do the check.